lib/rest-connector.js

```diff
@@ -100,31 +102,12 @@ function RestConnector(baseURL, settings) {
   this._settings = settings || {};
 
   // Extract the default options for request module
-  var options = this._settings.options ||
-    this._settings.defaults || this._settings;
+  var options = omit(defaults(this._settings.options || {},
+                              this._settings,
+                              this._settings.defaults),
+                     ['defaults', 'connector', 'operations']);
```

Comment:
defaults/options should have the precedence. We only merge settings if defaults/options is not present.